pkg/kademlia/kademlia.go

```diff
@@ -168,13 +169,14 @@ func (k *Kademlia) Bootstrap(ctx context.Context) (err error) {
 		var foundOnlineBootstrap bool
 		for i, node := range k.bootstrapNodes {
 			if ctx.Err() != nil {
+				k.log.Debug("Context Error received while Boostraping ", zap.Stringer(k.routingTable.self.Type.String(), k.routingTable.self.Id), zap.Stringer("Bootstrap Node", node.Id))
 				errGroup.Add(ctx.Err())
 				return errGroup.Err()
 			}
 
 			ident, err := k.dialer.FetchPeerIdentityUnverified(ctx, node.Address.Address)
 			if err != nil {
-				errGroup.Add(BootstrapErr.Wrap(err))
+				errGroup.Add(BootstrapErr.Wrap(BootstrapErr.New("%s : %s unable to fetch unverified peer identity node address %s: %s", k.routingTable.self.Type.String(), k.routingTable.self.Id.String(), node.Address.Address, err)))
```

Comment:
BootstrapErr.New already wraps internally, so no need to do that explicitly. Also there's no need to log self.